src/lib/server/state.c

```diff
@@ -687,9 +687,9 @@ int fr_request_to_state(fr_state_tree_t *state, request_t *request)
 	request_data_list_init(&data);
 	request_data_by_persistance(&data, request, true);
 
-	if (!request->session_state_pairs && fr_dlist_empty(&data)) return 0;
+	if (fr_pair_list_empty(&request->session_state_pairs) && fr_dlist_empty(&data)) return 0;
 
-	if (request->session_state_pairs) {
+	if (!fr_pair_list_empty(&request->session_state_pairs)) {
```

Comment:
This doesn't seem right.  `request->session_state_pairs` is a pointer to a list head, shouldn't need `&`?

Reply:
`session_state_pairs` is defined as `pair_list.state->children`, making `request->session_state_pairs` `request->pair_list.state->children` which is an `fr_pair_list_t` (not a pointer)

Reply:
Yes I realised this later.  Sorry I thought i'd removed these.